tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -281,6 +292,25 @@ public static void InstallPlaywrightBrowser()
                 throw new Exception($"Playwright exited with code {exitCode}");
             }
         }
+
+        /// <summary>
+        /// Requests a secret from keyvault using the default azure credentials
+        /// </summary>
+        /// <param name="keyvaultUri">The URI including path to the secret directory in keyvault</param>
+        /// <param name="keyvaultSecretName">The name of the secret</param>
+        /// <returns>The value of the secret from key vault</returns>
+        /// <exception cref="ArgumentNullException">Throws if no secret name is provided</exception>
+        internal static async Task<string> GetValueFromKeyvaultWitDefaultCreds(Uri keyvaultUri, string keyvaultSecretName)
+        {
+            if (string.IsNullOrEmpty(keyvaultSecretName))
+            {
+                throw new ArgumentNullException(nameof(keyvaultSecretName));
+            }
+
+            SecretClient client = new(keyvaultUri, new DefaultAzureCredential());

```

Comment:
You might not want to instanciate a DefaultAzureCredential each time? It would be better to have it as a static member

Reply:
done